File: backend/src/schemas/hourClass.ts
import { date, integer, pgTable, serial, timestamp } from 'drizzle-orm/pg-core'
import { createInsertSchema, createSelectSchema } from 'drizzle-zod'
import { classResourceTable } from './classResource'
import { hourTable } from './hour'

export const hourClassTable = pgTable('hour_class', {
  id: serial('id').primaryKey(),
  hourId: integer('hour_id').notNull().references(() => hourTable.id),
  classResourceId: integer('class_resource_id').references(() => classResourceTable.id),
  date: date('date').notNull(),
  createdAt: timestamp('created_at', { withTimezone: true }).notNull().defaultNow(),
  updatedAt: timestamp('updated_at', { withTimezone: true }).notNull().defaultNow()
})

export const selectHourClassSchema = createSelectSchema(hourClassTable)
export const insertHourClassSchema = createInsertSchema(hourClassTable)

File: flutter_frontend/lib/classes/hour_class.dart
class HourClass {
  final int? id;
  final int hourId;
  final int classResourceId;
  final String date;
  final String createdAt;
  final String updatedAt;

  HourClass({
    this.id,
    required this.hourId,
    required this.classResourceId,
    required this.date,
    required this.createdAt,
    required this.updatedAt,
  });

  factory HourClass.fromJson(Map<String, dynamic> json) {
    return switch (json) {
      {
        "id": int id,
        "hourId": int hourId,
        "classResourceId": int classResourceId,
        "date": String date,
        "createdAt": String createdAt,
        "updatedAt": String updatedAt,
      } =>
        HourClass(
          id: id,
          hourId: hourId,
          classResourceId: classResourceId,
          date: date,
          createdAt: createdAt,
          updatedAt: updatedAt,
        ),
      _ => throw const FormatException("HourClass: wrong format"),
    };
  }
}
